docs/website/versioned_docs/version-3.0.0/command-reference/delete.md

```diff
@@ -1,40 +1,39 @@
 ---
-title: odo delete
+title: odo delete component
 sidebar_position: 4
 ---
 
-`odo delete` command is useful for deleting resources that are managed by odo.
+`odo delete component` command is useful for deleting resources that are managed by odo. It deletes the component and it's related innerloop and outerloop resources from the cluster.
 
-## Deleting a component
-
-To delete a _devfile_ component, you can execute `odo delete`.
+There are 2 ways to deleting a component:
+1. [Delete with access to Devfile](/#Delete with access to Devfile)
+2. [Delete without access to Devfile](/#Delete without access to Devfile)
 
+## Delete with access to Devfile
 ```shell
-odo delete
+odo delete component
 ```
+odo analyzes the devfile for the devfile component and the outerloop resources.
+If the component has been deployed on the cluster, then odo will list all the resources and prompt the user to confirm the deletion.
+Otherwise, odo will exit with a message stating that it could not find the resources on the cluster.
 
-If the component is pushed to the cluster, running the above command will delete the component from the cluster, and it's dependant storage, url, secrets, and other resources.
-If it is not pushed, the command would exit with an error stating that it could not find the resources on the cluster.
-
-Use `-f` or `--force` flag to avoid the confirmation questions. 
-
-## Un-deploying Devfile Kubernetes components
+_Note:_ odo does not delete the Devfile, and config files.
 
-To undeploy the Devfile Kubernetes components deployed with `odo deploy` from the cluster, you can execute the `odo delete` command with `--deploy` flag:
+## Delete without access to Devfile
 ```shell
-odo delete --deploy
+odo delete component --name <component_name> --namespace <namespace>
```

Comment:
In this command, `--namespace` is an optional flag. Please explain so. Looks like we have done that well in the CLI docs.